File: tps/tp01/inversao_strings2.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool stop(char *input) {
    bool resp = false;
    
    if (strlen(input) == 3 && input[0] == 'F' && input[1] == 'I' && input[2] == 'M') {
        resp = true;
    }
    
    return resp;
}

void inverteString(char *input, char *resultado) {
    int n = strlen(input);
    
    for (int i = 0; i < n; i++) {
        resultado[i] = input[n-1-i];
    }
    resultado[n] = '\0';
}

int main() {
    char input[1000];
    char resultado[1000];
    
    fgets(input, 1000, stdin);
    input[strcspn(input, "\n")] = '\0';
    
    while (!stop(input)) {
        inverteString(input, resultado);
        printf("%s\n", resultado);
        
        fgets(input, 1000, stdin);
        input[strcspn(input, "\n")] = '\0';
    }
    
    return 0;
}

File: tps/tp01/inversao_strings2.java
import java.util.Scanner;


public class inversao_strings2 {
    
    public static boolean stop(String input) {
        boolean resp = false;

        if (input.length() == 3 && input.charAt(0) == 'F' && input.charAt(1) == 'I' && input.charAt(2) == 'M') 
        {
            resp = true;
        }

        return resp;}

    public static String inverteString(String input){
        int n = input.length();
        char ArrayPalavra[] = new char[1000];
        String resultado = "";
        for(int i = 0; i < n; i++){
            ArrayPalavra[i] = input.charAt(n-1-i);
            resultado += ArrayPalavra[i];
        }
        return resultado;
    
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        while(!stop(input)){
            System.out.println(inverteString(input));
            input = scanner.nextLine();
        }
        scanner.close();
    }
}
